core/frontend/src/quantity-formatting/QuantityFormatter.ts

```diff
@@ -551,8 +551,15 @@ export class QuantityFormatter implements UnitsProvider {
   /** async method to set a units provider and reload caches */
   public async setUnitsProvider(unitsProvider: UnitsProvider) {
     this._unitsProvider = unitsProvider;
-    // force all cached data to be reinitialized
-    await IModelApp.quantityFormatter.onInitialized();
+
+    try {
+      // force all cached data to be reinitialized
+      await IModelApp.quantityFormatter.onInitialized();
+    } catch(_) {
+      // If there is a problem initializing with the given provider, default back to the internal provider
+      await IModelApp.quantityFormatter.resetToUseInternalUnitsProvider();
```

Comment:
Is it possible to add some kind of warning here that says we are falling back on the internal provider because the imodel does not have the units schema imported

Reply:
I added a log message.